// analisadorlexicosintatico/dist/constants.js
"use strict";
exports.__esModule = true;
var POS_ESQUERDA = 'Esquerda';
exports.POS_ESQUERDA = POS_ESQUERDA;
var POS_DIREITA = 'Direita';
exports.POS_DIREITA = POS_DIREITA;
var POS_CENTRO = 'Centro';
exports.POS_CENTRO = POS_CENTRO;
var TIPO_NAO_TERMINAL = 'NÃO TERMINAL';
exports.TIPO_NAO_TERMINAL = TIPO_NAO_TERMINAL;
var TIPO_TERMINAL = 'TERMINAL';
exports.TIPO_TERMINAL = TIPO_TERMINAL;
var TIPO_E = 'E';
exports.TIPO_E = TIPO_E;
var TIPO_NUMERO = 'NÚMERO';
exports.TIPO_NUMERO = TIPO_NUMERO;
var TIPO_INCOGNITA = 'INCÓGNITA';
exports.TIPO_INCOGNITA = TIPO_INCOGNITA;
var TIPO_OPERADOR = 'OPERADOR';
exports.TIPO_OPERADOR = TIPO_OPERADOR;
var TIPO_ABRE_PARENTESES = '(';
exports.TIPO_ABRE_PARENTESES = TIPO_ABRE_PARENTESES;
var DESC_TIPO_ABRE_PARENTESES = 'ABRE PARÊNTESES';
exports.DESC_TIPO_ABRE_PARENTESES = DESC_TIPO_ABRE_PARENTESES;
var TIPO_FECHA_PARENTESES = ')';
exports.TIPO_FECHA_PARENTESES = TIPO_FECHA_PARENTESES;
var DESC_TIPO_FECHA_PARENTESES = 'FECHA PARÊNTESES';
exports.DESC_TIPO_FECHA_PARENTESES = DESC_TIPO_FECHA_PARENTESES;
var ARR_OPERADORES = ['+', '-', '/', '*'];
exports.ARR_OPERADORES = ARR_OPERADORES;
var ARR_INCOGNITAS = ["x", "y"];
exports.ARR_INCOGNITAS = ARR_INCOGNITAS;
