// analisadorlexicosintatico/dist/lexico.js
"use strict";
exports.__esModule = true;
var constants_1 = require("./constants");
function analisadorLexico(txt_entrada) {
    var array_tokens = txt_entrada.split(" ");
    var tipo_token = '-';
    var posicao_token = 0;
    var lista_tokens = [];
    for (var _i = 0, array_tokens_1 = array_tokens; _i < array_tokens_1.length; _i++) {
        var token = array_tokens_1[_i];
        tipo_token = '-';
        if (eh_incognita(token)) {
            tipo_token = constants_1.TIPO_INCOGNITA;
        }
        else if (eh_operador(token)) {
            tipo_token = constants_1.TIPO_OPERADOR;
        }
        else if (eh_numero(token)) {
            tipo_token = constants_1.TIPO_NUMERO;
        }
        else if (eh_abre_parentese(token)) {
            tipo_token = constants_1.DESC_TIPO_ABRE_PARENTESES;
        }
        else if (eh_fecha_parentese(token)) {
            tipo_token = constants_1.DESC_TIPO_FECHA_PARENTESES;
        }
        lista_tokens.push({ token: token, identificador: tipo_token, posicao: posicao_token });
        posicao_token += 1;
    }
    return lista_tokens;
}
exports.analisadorLexico = analisadorLexico;
function eh_operador(token) {
    return constants_1.ARR_OPERADORES.some(function (operador) { return operador === token; });
}
exports.eh_operador = eh_operador;
function eh_incognita(token) {
    return constants_1.ARR_INCOGNITAS.some(function (incognita) { return incognita === token; });
}
function eh_numero(token) {
    return !isNaN(token) && token !== "";
}
function eh_abre_parentese(token) {
    return token === constants_1.TIPO_ABRE_PARENTESES;
}
exports.eh_abre_parentese = eh_abre_parentese;
function eh_fecha_parentese(token) {
    return token === constants_1.TIPO_FECHA_PARENTESES;
}
exports.eh_fecha_parentese = eh_fecha_parentese;
