// analisadorlexicosintatico/dist/classes/infoNode.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var constants_1 = require("../constants");
var InfoNode = (function () {
    function InfoNode(valor, tipo) {
        this.valor = valor;
        this.tipo = tipo;
    }
    return InfoNode;
}());
exports.InfoNode = InfoNode;
var NaoTerminal = (function (_super) {
    __extends(NaoTerminal, _super);
    function NaoTerminal(valor) {
        return _super.call(this, valor, constants_1.TIPO_NAO_TERMINAL) || this;
    }
    return NaoTerminal;
}(InfoNode));
exports.NaoTerminal = NaoTerminal;
var E = (function (_super) {
    __extends(E, _super);
    function E() {
        return _super.call(this, constants_1.TIPO_E) || this;
    }
    return E;
}(NaoTerminal));
exports.E = E;
var Numero = (function (_super) {
    __extends(Numero, _super);
    function Numero() {
        return _super.call(this, constants_1.TIPO_NUMERO) || this;
    }
    return Numero;
}(NaoTerminal));
exports.Numero = Numero;
var Incognita = (function (_super) {
    __extends(Incognita, _super);
    function Incognita() {
        return _super.call(this, constants_1.TIPO_INCOGNITA) || this;
    }
    return Incognita;
}(NaoTerminal));
exports.Incognita = Incognita;
var Operador = (function (_super) {
    __extends(Operador, _super);
    function Operador() {
        return _super.call(this, constants_1.TIPO_OPERADOR) || this;
    }
    return Operador;
}(NaoTerminal));
exports.Operador = Operador;
var Terminal = (function (_super) {
    __extends(Terminal, _super);
    function Terminal(valor) {
        return _super.call(this, valor, constants_1.TIPO_TERMINAL) || this;
    }
    return Terminal;
}(InfoNode));
exports.Terminal = Terminal;
var AbreParenteses = (function (_super) {
    __extends(AbreParenteses, _super);
    function AbreParenteses() {
        return _super.call(this, constants_1.TIPO_ABRE_PARENTESES) || this;
    }
    return AbreParenteses;
}(Terminal));
exports.AbreParenteses = AbreParenteses;
var FechaParenteses = (function (_super) {
    __extends(FechaParenteses, _super);
    function FechaParenteses() {
        return _super.call(this, constants_1.TIPO_FECHA_PARENTESES) || this;
    }
    return FechaParenteses;
}(Terminal));
exports.FechaParenteses = FechaParenteses;
