// analisadorlexicosintatico/dist/classes/tree.js
"use strict";
exports.__esModule = true;
var infoNode_1 = require("./infoNode");
var constants_1 = require("../constants");
var Tree = (function () {
    function Tree() {
        this.raiz = new Node(new infoNode_1.E());
    }
    return Tree;
}());
exports.Tree = Tree;
var Node = (function () {
    function Node(object) {
        this.esquerdo = null;
        this.centro = null;
        this.direito = null;
        this.info = null;
        this.info = new infoNode_1.InfoNode(object.valor, object.tipo);
    }
    Node.prototype.visitInOrdem = function (result) {
        if (result === void 0) { result = ''; }
        if (this.esquerdo !== null) {
            result = this.esquerdo.visitInOrdem(result);
        }
        if (this.centro !== null) {
            result = this.centro.visitInOrdem(result);
        }
        if (this.direito !== null) {
            result = this.direito.visitInOrdem(result);
        }
        if (this.info.tipo === constants_1.TIPO_TERMINAL) {
            result = result + this.info.valor + " ";
        }
        return result;
    };
    Node.prototype.getJSON = function () {
        var filhoAtual = { text: { name: this.info.valor }, children: [] };
        if (this.direito !== null) {
            filhoAtual.children.unshift(this.direito.getJSON());
        }
        if (this.centro !== null) {
            filhoAtual.children.unshift(this.centro.getJSON());
        }
        if (this.esquerdo !== null) {
            filhoAtual.children.unshift(this.esquerdo.getJSON());
        }
        return filhoAtual;
    };
    return Node;
}());
exports.Node = Node;
